crates/hir_ty/src/display.rs

```diff
@@ -390,11 +418,27 @@ impl HirDisplay for ApplicationTy {
                 };
                 let trait_ = f.db.trait_data(trait_);
                 let type_alias = f.db.type_alias_data(type_alias);
-                write!(f, "{}::{}", trait_.name, type_alias.name)?;
-                if self.parameters.len() > 0 {
-                    write!(f, "<")?;
-                    f.write_joined(&*self.parameters.0, ", ")?;
-                    write!(f, ">")?;
+
+                // Use placeholder associated types when the target is test (https://rust-lang.github.io/chalk/book/clauses/type_equality.html#placeholder-associated-types)
+                if f.display_target.is_test() || self.parameters.len() > 1 {
```

Comment:
hm, I don't quite understand the logic here. why the `self.parameters.len() > 1`? We should still display a projection if there are more type parameters.

Reply:
Do you have an example of projection with more than 1 type parameters please ? Just to check the syntax :)

Reply:
The rest of the type parameters are the parameters of the trait (right now, as we don't have generic associated types). So it'd look like `<T as Foo<A, B, C>>::Type`. Actually, I think you could just reuse the existing code though -- just construct a `ProjectionTy` out of the `type_alias` and `parameters`, and call its `hir_fmt` method :slightly_smiling_face:

Reply:
Thanks for your help ! Updated